server/driver/wivrn_controller.cpp

```diff
@@ -448,8 +452,18 @@ void wivrn_controller::update_tracking(const from_headset::tracking & tracking,
 		cnx->set_enabled(aim.device, false);
 	if (not grip.update_tracking(tracking, offset))
 		cnx->set_enabled(grip.device, false);
-	if (not palm.update_tracking(tracking, offset))
+
+	bool palm_tracked = false;
+	// FIXME: The offset should be based off of the headset instead
+	if (cnx->get_info().palm_pose == false || configuration::read_user_configuration().force_palm) {
```

Comment:
`read_user_configuration` parses the json, we don't want to do that on every packet from the network!

The offset should be stored in the `pose_list` object at construction so that you can use it normally after.

Reply:
Kk, do you want the offset stored in pose list to be just a vec3 or also an optional so not rotating quats for ones that dont need it?